pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -446,19 +506,21 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 		if rawPodInfo == nil {
 			break
 		}
-		pod := rawPodInfo.(*framework.QueuedPodInfo).Pod
-		if p.isPodBackingoff(rawPodInfo.(*framework.QueuedPodInfo)) {
+		pInfo := rawPodInfo.(*framework.QueuedPodInfo)
+		pod := pInfo.Pod
+		if p.isPodBackingoff(pInfo) {
 			break
 		}
 		_, err := p.podBackoffQ.Pop()
 		if err != nil {
 			klog.ErrorS(err, "Unable to pop pod from backoff queue despite backoff completion", "pod", klog.KObj(pod))
 			break
 		}
-		p.activeQ.Add(rawPodInfo)
-		klog.V(5).InfoS("Pod moved to an internal scheduling queue", "pod", klog.KObj(pod), "event", BackoffComplete, "queue", activeQName)
-		metrics.SchedulerQueueIncomingPods.WithLabelValues("active", BackoffComplete).Inc()
-		activated = true
+		if added, _ := p.addToActiveQ(pInfo); added {
```

Comment:
Will gated pod possible in backoffQ? Related to the question I asked here https://github.com/kubernetes/kubernetes/pull/113275/files#r1013651169

Reply:
It's technically impossible for a gated pod to stay in backoffQ.

Reply:
Then `p.activeQ.Add` seems enough. For we only support one-direction state transition from gated -> non-gated, and pods in backoffQ are all non-gated.

Reply:
that's a fair point. Let me record it as a follow-up in #113608.